core/src/test/java/org/elasticsearch/index/query/InnerHitBuilderTests.java

```diff
@@ -221,7 +221,12 @@ public static InnerHitBuilder randomInnerHits(boolean recursive, boolean include
         innerHits.setTrackScores(randomBoolean());
         innerHits.setFieldNames(randomListStuff(16, () -> randomAsciiOfLengthBetween(1, 16)));
         innerHits.setFieldDataFields(randomListStuff(16, () -> randomAsciiOfLengthBetween(1, 16)));
-        innerHits.setScriptFields(new HashSet<>(randomListStuff(16, InnerHitBuilderTests::randomScript)));
+        // Random script fields deduped on their field name.
+        Map<String, SearchSourceBuilder.ScriptField> scriptFields = new HashMap<>();
```

Comment:
And again, looking at the code, they *do* use the field name in equality, but for the query, you can't have multiple *different* queries even with the same name (which is why the hashset doesn't work), I'll stop confusing myself now.